### PR TITLE
Eliminate BITVECTOR_ITE in bv_to_bool preprocessing pass

### PR DESCRIPTION
**Pull request by yoni206**
Currently the `bv_to_bool` preprocessing pass does not eliminate `BITVECTOR_ITE` nodes.
However, in a future PR that will introduce the new `bv_to_int` preprocessing pass, it is convenient for me to assume that such nodes do not appear.

This PR translates `BITVECTOR_ITE` nodes into "regular" `ITE` nodes.

While working on a unit test for this feature, I noticed that the unit test for the `bv_gauss` preprocessing pass does not run with `make units`. This PR fixes that as well.

### CODE REVIEW

File: test/unit/preprocessing/pass_bv_to_bool_white.h [hunk 1]
```diff
@@ -0,0 +1,86 @@
+/*********************                                                        */
+/*! \file pass_bv_to_bool_white.h
+ ** \verbatim
+ ** Top contributors (to current version):
+ **   Aina Niemetz, Andres Noetzli
```
Comment (mpreiner) [inline test/unit/preprocessing/pass_bv_to_bool_white.h]: Shouldn't that be Yoni Zohar?

Comment (yoni206) [reply test/unit/preprocessing/pass_bv_to_bool_white.h]: correct, fixed.

File: src/preprocessing/passes/bv_to_bool.cpp [hunk 2]
```diff
@@ -171,9 +169,15 @@ Node BVToBool::convertBvTerm(TNode node)
...
-  if (kind == kind::ITE)
+  if (kind == kind::ITE || kind == kind::BITVECTOR_ITE)
   {
-    Node cond = liftNode(node[0]);
+    Node cond = node[0];
+    // Translate bit-vector condition to an equality to (_ bv1 1)
+    if (kind == kind::BITVECTOR_ITE)
+    {
+      cond = NodeManager::currentNM()->mkNode(kind::EQUAL, cond, d_one);
+    }
```
Comment (mpreiner) [inline src/preprocessing/passes/bv_to_bool.cpp]: What happens if we have e.g.,

```
(BITVECTOR_ITE (BITVECTOR_COMP ...) x y)
```

`BITVECTOR_COMP` gets converted to `EQUAL`. Can it happen that you will compare an `EQUAL` against  `d_one`?

**Comment by yoni206** [comment]
Eliminating `BITVECTOR_ITE` is possible, but a bit more complex than the other operators, as the type of `BITVECTOR_ITE` terms may be different than `(_ BitVec 1)`. Since the only known application of this elimination is my future preprocessing pass, I will just implement it there.

**Pull request closed without merging**